### ISSUE TITLE
New program runs over the last one after reset

### ISSUE THREAD

**Issue by JoshuaLowe1002** [microbit-foundation/micropython-microbit-v2-simulator#90]
When a reset event is sent to the simulator, it seems a new program is running over the last one from before the reset. Can be seen best when a message is scrolling on the display, my best guess is that this is related to the changes made in #88.

Example from the Python Editor:

https://user-images.githubusercontent.com/25752941/199282702-4416f5f2-f6b2-4d82-94bb-2ed80276dc1d.mov

**Comment by microbit-matt-hillsdon** [comment]
Thanks. Really easily seen with a program like this:

```python
from microbit import *
import random

r = random.randrange(0, 100)
while True:
    print(r)
    sleep(1000)
```

I'll look at this soon. We might temporarily revert #88 if it looks non-trivial to fix.

**Comment by microbit-matt-hillsdon** [comment]
There are two places where we `await stop` since #88 but this isn't safe as we're really only waiting for stop to be requested. I'll revert #88 for the moment and try to rework this later.

**Comment by microbit-matt-hillsdon** [comment]
I've reverted #88 and released it as 0.1.7. Confirmed fixed in the micro:bit Python Editor V3 with the program in my comment above. I'll look again at #86 soon.

Thanks for reporting @JoshuaLowe1002

**Comment by microbit-matt-hillsdon** [comment]
For info, there's a new attempt at fixing #86 without introducing this issue on #91 that we'll be testing today.

You can try it out at 
https://review-python-editor-v3.microbit.org/user-stop-wins/ and
https://review-python-simulator.usermbit.org/user-stop-wins/demo.html
